tools/scrape/vimorg.py

```diff
@@ -23,25 +24,67 @@ def __init__(self, *args, **kwargs):
 PARSER = HTMLParser()
 
 
-def scrape_scripts(num):
-    """Scrapes and upserts the num most recently created scripts on vim.org."""
+def get_all_info_from_url_and_name(vimorg_url, name):
+    vimorg_data = urlparse.urlparse(vimorg_url)
+    query_string = urlparse.parse_qs(vimorg_data.query)
+    if not query_string or 'script_id' not in query_string:
+        return {}
+
+    vimorg_id = query_string['script_id'][0]
     res = requests.get(
-            'http://www.vim.org/scripts/script_search_results.php?show_me=%d' %
-                num)
+        'http://www.vim.org/scripts/script_search_results.php?keywords=%s' %
+        name)
+
+    plugins = get_plugins_info_from_response(res)
+    for plugin in plugins:
+        if str(plugin['vimorg_id']) == str(vimorg_id):
+            return plugin
+    return None
 
+
+def get_plugins_info_from_response(res):
     html = lxml.html.html5parser.document_fromstring(res.text, parser=PARSER)
 
     # Since there are no identifying classes or ids, this is the best way to
     # find the correct table
     scripts = html.xpath('//table[tbody/tr/th[contains(text(),"Script")]]/*/*')
 
+    plugins = []
     # the first two rows and the last row aren't scripts
     for tr in scripts[2:-1]:
         link = tr[0][0].attrib['href']
 
-        vimorg_id = re.search("script_id=(\d+)", link).group(1)
-        name = tr[0][0].text.encode('utf-8')
```

Comment:
I wonder why there's an `encode('utf-8')` here but not for the other fields like `vimorg_type`, for example, further below ...

Reply:
No idea, I just didn't want to break it.